### PR TITLE
Fix some of the minor typos found while reading the docs

### PR DESCRIPTION
**Pull request by RubenRBS**
These are some of the low hanging typos I've found today. I've committed them mostly to check that my setup is ok, and to learn workflow for these cases.

Should I now check if they are present in develop2 and merge there too? Or is the process different?

### CODE REVIEW

File: devtools/build_requires.rst [hunk 1]
```diff
@@ -262,7 +262,7 @@ Making tool_requires affect the consumers package-ID
     This subsection should be considered a workaround, not a feature, and it might have other side effects, that will not be fixed
     as this is not recommended production code.
 
-As discussed above, the ``tool_requires`` do not affect at all the package ID. As they will not be present at all when the ``package_id``
+As discussed above, the ``tool_requires`` does not affect at all the package ID. As they will not be present at all when the ``package_id``
```
Comment (memsharded) [inline devtools/build_requires.rst:265]:
```suggestion
As discussed above, the ``tool_requires`` do not affect at all the package ID. As they will not be present at all when the ``package_id``
```

I think this is good, it means plural, there can be several ``tool_requires``, no one of them affects package_id. The plural is also implicit in later ``As they will not be present...``

File: devtools/build_requires.rst [hunk 2]
```diff
@@ -306,7 +306,7 @@ as a ``tool_requires`` or ``test_requires``:
             self.test_requires(self.tested_reference_str)
 
 
-If for some reason, it is necessary to test the same package both as a regular require and a build_require,
+If for some reason, it is necessary to test the same package both as a regular require and a tool_require,
```
Comment (memsharded) [inline devtools/build_requires.rst:309]:
```suggestion
If for some reason, it is necessary to test the same package both as a regular require and a ``tool_require``,
```

**Review by memsharded — APPROVED** [review]

File: devtools/build_requires.rst (hunk 1 above)
Comment (RubenRBS) [reply devtools/build_requires.rst]: You're right, sorry! Nice catch

**Review by memsharded — APPROVED** [review]

**Comment by memsharded** [comment]
Some other hints:
- If you target the ``master`` branch, merged changes will go to the live docs "latest"
- Changes to ``develop`` branch goes to the next release docs (that will become latest)

Docs in Conan 2.0 are completely different, written from scratch. I don't think looking for 1:1 correspondence of those typos is worth, but feel free to do a quick check.

**Pull request merged**